android/src/com/mapswithme/maps/routing/RoutingPlanController.java

```diff
@@ -108,18 +127,29 @@ public void onClick(View v)
       }
     });
 
+    setupRouterButton(R.id.taxi, R.drawable.ic_taxi, new View.OnClickListener()
+    {
+      @Override
+      public void onClick(View v)
+      {
+        AlohaHelper.logClick(AlohaHelper.ROUTING_TAXI_SET);
```

Comment:
We can call Aloha on the main thread even in core. My proposal is to notify aloha about taxi routing type from the core, not in the UI.